cmds.go

```diff
@@ -49,6 +72,75 @@ type SonyCmdToggle struct {
 	Input string `short:"i" help:"Specify host input, do not autodetect"`
 }
 
+// AfterApply creates a new [Screen] from the flags in the [screenFlags] struct.
+func (sf *screenFlags) AfterApply() error {
+	s, err := NewScreen(sf.Display, sf.Manufacturer, sf.ProductCode)
+	if err != nil {
+		return err
+	}
+	sf.screen = s
+	return nil
+}
+
+// Run (offscreen run) runs offscreen to turn the connected TV on and off
+// in line with X screen saver events.
+func (cmd *RunCmd) Run() error {
+	ch := make(chan bool)
+	g := errgroup.Group{}
+	g.Go(func() error { return cmd.screen.Watch(ch) })
+	g.Go(func() error { return cmd.loop(ch) })
+	return g.Wait()
+}
+
+func (cmd *RunCmd) loop(ch <-chan bool) error {
+	defer cmd.screen.Close()
+	c := NewRESTClient(cmd.Hostname, cmd.PSK)
+	ourLabel, err := hostnameLabel()
+	if err != nil {
+		return fmt.Errorf("could not get input URI for %s: %w", cmd.Input, err)
+	}
+	labels, err := c.Inputs()
+	if err != nil {
+		return err
+	}
+	ourInput, ok := labels[ourLabel]
+	if !ok {
+		return fmt.Errorf("tv set does not have labelled input: %s", ourLabel)
+	}
+	cmd.Input = ourInput
```

Comment:
wait, why do you override this? I thought this comes from the flag, what's the point of having it?

Reply:
yeah, thats dumb. I do rework this in the next commit and it all is handled differently. I can rework this.

Reply:
This ended up disappearing due to reordering of commits.